integration-test/test_dummy_data.py

```diff
@@ -0,0 +1,185 @@
+    )
+    obs_get_response = grpc_stub.GetObservations(obs_get_request)
+    assert obs_get_response.tsobs[0].tsid == dummy_timeseries_for_delete.id
+    assert len(obs_get_response.tsobs[0].obs) == 3
+
+    ts_delete_request = dstore.DeleteTSRequest(ids=[dummy_timeseries_for_delete.id])
+    ts_delete_response = grpc_stub.DeleteTimeSeries(ts_delete_request)
+    assert str(ts_delete_response) == "status: -1\n"
+
+    ts_find_response = grpc_stub.FindTimeSeries(ts_find_request)
+    assert len(ts_find_response.tseries) == 0
+
+    obs_get_response = grpc_stub.GetObservations(obs_get_request)
+    assert obs_get_response.tsobs[0].tsid == dummy_timeseries_for_delete.id
+    assert len(obs_get_response.tsobs[0].obs) == 0
+
+
+@pytest.fixture(scope="module")
+def setup_dummy_data_for_polygon(grpc_stub):
+    dummy_id = 999999990
+    dummy_str_id = "999999990"
+    ts_metadata = dstore.TSMetadata(
+        station_id=dummy_str_id,
+        param_id=dummy_str_id,
+        pos=dstore.Point(lat=80, lon=170),
+        other1="test_value1",
+        other2="test_value2",
+        other3="test_value3",
+    )
+    ts_add_request_1 = dstore.AddTSRequest(
+        id=dummy_id,
+        metadata=ts_metadata,
+    )
+    grpc_stub.AddTimeSeries(ts_add_request_1)
+
+    dummy_id = 999999991
+    dummy_str_id = "999999991"
+    ts_metadata = dstore.TSMetadata(
+        station_id=dummy_str_id,
+        param_id=dummy_str_id,
+        pos=dstore.Point(lat=90, lon=180),
+        other1="test_value4",
+        other2="test_value5",
+        other3="test_value6",
+    )
+    ts_add_request_2 = dstore.AddTSRequest(
+        id=dummy_id,
+        metadata=ts_metadata,
+    )
+    grpc_stub.AddTimeSeries(ts_add_request_2)
+
+    yield  # Hereafter teardown
+
+    grpc_stub.DeleteTimeSeries(dstore.DeleteTSRequest(ids=[ts_add_request_1.id, ts_add_request_2.id]))
+
+
+input_params_polygon = [
+    (
+        # Multiple stations within
+        ((80, 170), (90, 170), (90, 180), (80, 180)),
+        {"number_of_stations": 2, "station_ids": [999999990, 999999991]},
+    ),
+    (
+        # One station within
+        ((80, 175), (90, 175), (90, 180), (80, 180)),
+        {"number_of_stations": 1, "station_ids": [999999991]},
+    ),
+    (
+        # Nothing within
+        ((80, 175), (85, 175), (85, 180), (80, 180)),
+        {"number_of_stations": 0, "station_ids": []},
+    ),
+    (
+        # middle top
+        ((85, 120), (90, 120), (90, -120), (85, -120)),
+        {"number_of_stations": 1, "station_ids": [999999991]},
+    ),
+    (
+        # Middle bottom, should fall outside since polygon is curved because the earth is round (postgres geography).
+        ((85, 120), (80, 120), (85, -120), (80, -120)),
+        {"number_of_stations": 0, "station_ids": []},
+    ),
+    (
+        # Complex polygon
+        ((79, 169), (81, 169), (80, 171), (89, 171), (89, 180), (75, 180), (75, 171)),
+        {"number_of_stations": 1, "station_ids": [999999990]},
+    ),
+]
```

Comment:
Rewrite the tests away from the pole and the dateline. I am not sure these test what they say they do.
For example, (90, 180) == (80, 180).